frontend/src/App.js

```diff
@@ -438,7 +438,11 @@ const MenuPopup = ({user}) =>
             onClick={() => {
               close();
               bookSetState("user.uid", null)
-              firebase.auth().signOut();
+              try {
+                firebase.auth().signOut().catch(console.warn);
+              } catch (err) {
+                console.warn(err)
```

Comment:
May seem redundant, but the the `.catch()` is to handle the `.signOut()` promise, whereas the `try`/`catch` is to handle an error when invoking `firebase.auth()`.

(Same is true of a couple of other places in this changeset.)